addons/product/tests/test_pricelist.py

```diff
@@ -198,3 +198,35 @@ def test_pricelists_res_partner_form(self):
 
         with Form(partner) as partner_form:
             self.assertEqual(partner_form.property_product_pricelist, self.sale_pricelist_id)
+
+    def test_pricelist_change_to_formula_and_back(self):
+        pricelist_1 = self.env['product.pricelist'].create({
+            'name': 'Sale pricelist',
+            'item_ids': [
+                Command.create({
+                    'compute_price': 'fixed',
+                    'fixed_price': 40,
+                    'base': 'list_price',  # based on public price
+                    'price_discount': 10,
+                    'applied_on': '3_global',
+                }),
+            ],
+        })
+        pricelist_2 = self.env['product.pricelist'].create({
+            'name': 'Sale pricelist',
+            'item_ids': [
+                Command.create({
+                    'compute_price': 'percentage',
+                    'percent_price': 20,
+                    'base': 'pricelist',  # based on pricelist_1
+                    'base_pricelist_id': pricelist_1.id,
+                    'applied_on': '3_global',
+                }),
+            ],
+        })
```

Comment:
We can re-use the pricelist from `setUpClass` instead of creating 2 new ones.

```suggestion
    def test_pricelist_change_to_formula_and_back(self):
        pricelist_2 = self.env['product.pricelist'].create({
            'name': "Pricelist-based pricelist",
            'item_ids': [
                Command.create({
                    'compute_price': 'percentage',
                    'percent_price': 20,
                    'base': 'pricelist',
                    'base_pricelist_id': self.sale_pricelist_id.id,
                    'applied_on': '3_global',
                }),
            ],
        })
```